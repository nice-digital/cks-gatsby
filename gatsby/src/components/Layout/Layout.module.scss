@import '~@nice-digital/nds-core/scss/core';

@supports (position: sticky) {
  .main {
    position: relative; // So that the back to top link can dock
    padding-bottom: rem(
      $nds-spacing-medium * 4
    ); // Space for the back to top link + a gap
  }
}
