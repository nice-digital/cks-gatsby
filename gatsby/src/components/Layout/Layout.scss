$nds-include-spacing-modifiers: false;
$nds-output-text-classes: false;
$nds-output-visibility-classes: false;

@import '~@nice-digital/design-system/scss/base.scss';

// Markup components
@import './tables';

html {
  overflow-x: visible; // Override the overflow from NDS, otherwise position sticky doesn't work
  scroll-behavior: smooth;

  @media (prefers-reduced-motion) {
    scroll-behavior: auto;
  }
}

body {
  // We're using font-display for Google fonts loading so on a slow connection you see FOUT
  // Trebuchet is closer to Lato than Arial and supported on 97-99.7% of devices (https://www.cssfontstack.com/Trebuchet-MS)
  // so gives less of a jump when the font swaps.
  // We use Roboto for Android devices
  font-family: 'Lato', 'Trebuchet MS', 'Roboto', 'Helvetica Neue', Helvetica,
    Arial, sans-serif;
}

// $nds-output-visibility-classes stops visually-hidden being output
.visually-hidden {
  position: absolute;
  overflow: hidden;
  clip: rect(0 0 0 0);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
}

#content-start:focus {
  outline: none;
}
