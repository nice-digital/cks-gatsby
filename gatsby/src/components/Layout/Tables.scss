@import '~@nice-digital/nds-core/scss/core.scss';

// Taken from Niceorg Client

.table {
  width: 100%;
  border-collapse: collapse;
  th {
    vertical-align: bottom;
  }
  td {
    vertical-align: top;
  }
  td,
  th {
    text-align: left;
    padding: rem($nds-spacing-small);
    p {
      margin-top: 0;
    }
  }
  tr {
    &.success {
      background: $nds-colour-success-background;
    }
    &.error {
      background: $nds-colour-error-background;
    }
    &.warning {
      background: $nds-colour-caution-background;
    }
    &.info {
      background: $nds-colour-info-background;
    }
  }
  &-bordered {
    border: 1px solid $nds-colour-border;
    border-collapse: collapse;
    th,
    td {
      border: 1px solid $nds-colour-border;
    }
  }
  &-striped {
    tbody {
      tr:nth-child(odd) > td {
        background-color: $nds-colour-nice-grey-3;
      }
    }
  }
  &-hover {
    tbody {
      tr:hover > td {
        background-color: $nds-colour-nice-grey-3;
      }
    }
  }
  &-condensed {
    @include nds-font-size(-1);
    td,
    th {
      padding: rem($nds-spacing-x-small);
    }
  }
  &-responsive {
    @include mq($until: md) {
      width: 100%;
      overflow-y: hidden;
      overflow-x: scroll;
    }
  }
}
