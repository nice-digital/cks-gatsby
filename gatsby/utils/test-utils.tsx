import React, { ReactElement } from "react";
import { render, RenderResult } from "@testing-library/react";
import {
	createMemorySource,
	createHistory,
	LocationProvider,
	History,
} from "@reach/router";

// Util for rendering a component wrapped within Reach Router's LocationProvider
// See https://testing-library.com/docs/example-reach-router
export const renderWithRouter = (
	ui: ReactElement,
	{ route = "/", history = createHistory(createMemorySource(route)) } = {}
): { history: History } & RenderResult => ({
	...render(<LocationProvider history={history}>{ui}</LocationProvider>),
	// adding `history` to the returned utilities to allow us
	// to reference it in our tests (just try to avoid using
	// this to test implementation details).
	history,
});
