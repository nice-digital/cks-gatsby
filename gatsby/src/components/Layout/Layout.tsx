import React, { ReactNode } from "react";

import { Header } from "../Header/Header";
import { Footer } from "../Footer/Footer";
import { BackToTop } from "../BackToTop/BackToTop";

import "./Layout.scss";
import styles from "../BackToTop/BackToTop.module.scss";
import { Helmet } from "react-helmet";

type LayoutProps = {
	children: ReactNode;
};

export const Layout: React.FC<LayoutProps> = ({ children }: LayoutProps) => {
	return (
		<>
			<Helmet>
				<link
					rel="preconnect"
					href="https://fonts.gstatic.com"
					crossOrigin="crossOrigin"
				/>
				<link
					rel="stylesheet"
					href="https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,400;0,700;0,900;1,400&display=swap"
				/>
				<script
					src="https://cdn.nice.org.uk/cookie-banner/cookie-banner.min.js"
					type="text/javascript"
				></script>
			</Helmet>
			<Header />
			<main
				className={`container ${styles.backToTopSpacing}`}
				id="content-start"
			>
				{children}
			</main>
			<BackToTop />
			<Footer />
		</>
	);
};
