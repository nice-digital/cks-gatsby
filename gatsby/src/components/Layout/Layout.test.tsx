import React from "react";
import { renderWithRouter } from "test-utils";

import { Layout } from "./Layout";

describe("Layout", () => {
	it("should have skip link target id on main element", async () => {
		const { getByRole } = renderWithRouter(<Layout>Test</Layout>);

		expect(getByRole("main")).toHaveAttribute("id", "content-start");
	});
});
