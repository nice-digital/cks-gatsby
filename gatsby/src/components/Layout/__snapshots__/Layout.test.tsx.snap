// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match snapshot for EULA 1`] = `
<div
  class="panel eula"
  data-component="panel"
  data-testid="eula"
>
  <p>
    The content on the NICE Clinical Knowledge Summaries site (CKS) is the copyright of
     
    <a
      href="https://agiliosoftware.com/primary-care/"
      rel="noopener noreferrer"
    >
      Clarity Informatics Limited (trading as Agilio Software Primary Care)
    </a>
    . By using CKS, you agree to the licence set out in the
     
    <a
      href="https://www.nice.org.uk/terms-and-conditions/cks-end-user-licence-agreement"
    >
      CKS End User Licence Agreement
    </a>
    .
  </p>
</div>
`;
